sdk/core/azure-core/inc/http/policy.hpp

```diff
@@ -96,6 +97,8 @@ namespace Azure { namespace Core { namespace Http {
 
   class RequestIdPolicy : public HttpPolicy {
 
+    constexpr static const char* RequestIdHeader = "x-ms-request-id";
```

Comment:
It's actually `"x-ms-client-request-id"` for sending the request ID to server.
See this: https://docs.microsoft.com/en-us/rest/api/storageservices/get-blob-service-properties#request-headers